content/tutorials/android-push-server-registration.textile

```diff
@@ -0,0 +1,239 @@
+        @Override
+        public void onResponse(Call<NetResponse> call, Response<NetResponse> response) {
+            logMessage("Successfully registered: " + new Gson().toJson(response.body()));
+            handler.sendMessage(handler.obtainMessage(SUCCESS, STEP_3));
+        }
+
+        @Override
+        public void onFailure(Call<NetResponse> call, Throwable t) {
+            logMessage("Error registering with server: " + t.getMessage());
+            handler.sendMessage(handler.obtainMessage(FAILURE));
+        }
+    });
+}
+```
+
+
+In the 'server' folder that you created before, add a new file and name it @ServerAPI.java@. Here, we'll define the ServerAPI class that'll send a request to the server with the device registration info. Paste the following in the this file:
+
+```[java]
+package YOUR-PACKAGE-NAME;
+
+import io.ably.lib.push.AblyFirebaseInstanceIdService;
+/**
+ * Leave this empty as the base class AblyFirebaseInstanceIdService does the FCM token registration.
+ * In case your app requires access to FCM token as well, then override io.ably.lib.push.AblyFirebaseInstanceIdService#onTokenRefresh()
+ */
+public class AblyPushRegistrationService extends AblyFirebaseInstanceIdService {
+    @Override
+    public void onTokenRefresh() {
+        //Make sure to call super.onTokenRefresh to initialize Ably push environment.
+        super.onTokenRefresh();
+    }
+}
+```
+
+Next, in the same folder, that is 'server', add a new file and name it @AblyPushMessagingService.java@ and paste the following into it.
+
+```[java]
+package YOUR-PACKAGE-NAME;
+
+import io.ably.tutorial.push_tutorial_two.BuildConfig;
+import retrofit2.Call;
+import retrofit2.Retrofit;
+import retrofit2.converter.gson.GsonConverterFactory;
+import retrofit2.http.GET;
+import retrofit2.http.Query;
+
+public class ServerAPI {
+    public interface API {
+        @GET("/register")
+        Call<NetResponse> register(@Query("deviceId") String deviceId, @Query("registrationToken") String registrationToken, @Query("clientId") String clientId);
+    }
+
+
+    private static ServerAPI instance;
+
+    public static ServerAPI getInstance() {
+        if (instance == null) {
+            instance = new ServerAPI();
+        }
+        return instance;
+    }
+
+    private final API api;
+
+    private ServerAPI() {
+        Retrofit.Builder builder = new Retrofit.Builder();
+        builder.baseUrl(BuildConfig.ABLY_URL);
+        builder.addConverterFactory(GsonConverterFactory.create());
+
+        Retrofit retrofit = builder.build();
+        api = retrofit.create(API.class);
+    }
+
+    public API api() {
+        return api;
+    }
+}
+```
+
+If you observe, we have made use of the @NetResponse@ class. Let's go ahead and define this next, in the same file, i.e, 'server'. Create a new file named @NetResponse.java@ and paste the following into it:
```

Comment:
`If you observe, we have made use of the @NetResponse@ class` would be better just as `This makes use of @NetResponse@ class` IMO.